src/util/unicode.cpp

```diff
@@ -315,3 +318,12 @@ std::string utf16_native_endian_to_java(const std::wstring &in)
     utf16_native_endian_to_java(ch, result, loc);
   return result.str();
 }
+
+std::string utf16_hex_to_utf8(const std::string &hex)
+{
+  PRECONDITION(hex.length() == 4);
+  const char16_t utf16_char = std::strtol(hex.c_str(), nullptr, 16);
+  const std::u16string u16(1, utf16_char);
+  return std::wstring_convert<std::codecvt_utf8_utf16<char16_t>, char16_t>{}
```

Comment:
This is failing with a linker error with Visual Studio. It's a known problem and https://stackoverflow.com/questions/32055357/visual-studio-c-2015-stdcodecvt-with-char16-t-or-char32-t discusses some fixes. It seems that using `wchar_t` instead of `char16_t` might be the easiest of those, if that does actually work. But it would need to be wrapped in an `#ifdef _MSC_VER`.

Reply:
I tried the solution that uses `wchar_t`, the AWS Windows build is still failing and I can't tell from the logs if it's because of this or for some other reason... Can you tell from the logs what the problem is?

Reply:
Turns out the problem was a compilation error that, for some reason, did not show up anywhere in the AWS build logs.
Now everything builds fine but test are failing - again only on Windows...

Reply:
The problem was the encoding of string literals, which can vary between compilers. Should all be fixed now!